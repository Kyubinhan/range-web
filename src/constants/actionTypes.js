export const STORE_SSO_AUTH_DATA = 'STORE_SSO_AUTH_DATA';
export const STORE_USER = 'STORE_USER';
export const SIGN_OUT = 'SIGN_OUT';

export const REQUEST = 'REQUEST';
export const SUCCESS = 'SUCCESS';
export const ERROR = 'ERROR';
export const SUCCESS_PAGINATED = 'SUCCESS_PAGINATED';

export const STORE_AGREEMENTS = 'STORE_AGREEMENTS';
export const SET_AGREEMENT_FILTER = 'SET_AGREEMENT_FILTER';

export const STORE_PLAN = 'STORE_PLAN';
// export const STORE_ALBUM = 'STORE_ALBUM';
// export const STORE_ARTIST = 'STORE_ARTIST';
// export const STORE_LIBRARY_ALBUMS = 'STORE_LIBRARY_ALBUMS';
// export const FAVORITE_ALBUM = 'FAVORITE_ALBUM';
// export const UNFAVORITE_ALBUM = 'UNFAVORITE_ALBUM';

// export const SET_FILTER = 'SET_FILTER';
// export const SELECT_ARTIST = 'SELECT_ARTIST';
