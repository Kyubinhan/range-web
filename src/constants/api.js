export const SSO_BASE_URL = "https://dev-sso.pathfinder.gov.bc.ca/";
export const SSO_REALM_NAME = "mobile";
export const SSO_BASE_AUTH_ENDPOINT = `${SSO_BASE_URL}auth/realms/mobile/protocol/openid-connect/auth`;
export const SSO_CLIENT_ID = "range-test";
export const SSO_REDIRECT_URI = `${window.location.origin}/login`;

export const SSO_AUTH_ENDPOINT = SSO_BASE_AUTH_ENDPOINT
  + "?response_type=code"
  + `&client_id=${SSO_CLIENT_ID}`
  + `&redirect_uri=${SSO_REDIRECT_URI}`;

export const GET_TOKEN = `auth/realms/${SSO_REALM_NAME}/protocol/openid-connect/token`;
export const REFRESH_TOKEN = `auth/realms/${SSO_REALM_NAME}/protocol/openid-connect/token`

export const BASE_URL = 'http://api-range-myra-dev.pathfinder.gov.bc.ca/v1';
export const AGREEMENT = '/agreement';
export const REFERENCES = '/reference';
export const CLIENT = '/client';
export const STATUS = '/status';
export const ZONE = '/zone';