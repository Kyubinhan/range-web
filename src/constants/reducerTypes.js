export const AUTH = 'AUTH';
export const TOAST = 'TOAST';
export const GET_RANGE_USE_PLANS = 'GET_RANGE_USE_PLANS';
