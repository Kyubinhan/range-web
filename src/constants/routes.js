export const HOME = '/home';
export const LOGIN = '/login';
export const LOGOUT = '/logout';

export const RANGE_USE_PLAN = '/range-use-plan';
export const RANGE_USE_PLANS = '/range-use-plans';

export const MANAGE_ZONE = '/manage-zone';
export const EXPORT_PDF = '/export-pdf';
