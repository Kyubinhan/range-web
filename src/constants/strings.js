// tenure agreement table header labels
export const RANGE_NUMBER = 'RAN Number';
export const RANGE_NAME = 'Range Name';
export const AGREEMENT_HOLDER = 'Agreement Holder';
export const STATUS = 'Status';
export const STAFF_CONTACT = 'Staff Contact';

// Range use plan basic information
export const PLAN_START = 'Plan Start Date';
export const PLAN_END = 'Plan End Date';
export const AGREEMENT_START = 'Agreement Start Date';
export const AGREEMENT_END = 'Agreement End Date';
export const AGREEMENT_DATE = 'Agreement Date';
export const AGREEMENT_TYPE = 'Agreement Type';
export const DISTRICT = 'District (Responsible)';
export const ZONE = 'Zone';
export const ALTERNATIVE_BUSINESS_NAME = 'Alternative Business Name';
export const AGREEMENT_HOLDERS = 'Agreement Holders';
export const TYPE = 'Type';
export const NOT_PROVIDED = 'Not provided';
export const NP = 'N/P';
export const NOT_SELECTED = 'Not selected';
export const NO_RUP_PROVIDED = 'No RUP found';
export const CONTACT_NAME = 'Contact Name';
export const CONTACT_EMAIL = 'Contact Email';
export const CONTACT_PHONE = 'Contact Phone';
export const EXTENDED = 'Extended';
export const EXEMPTION_STATUS = 'Exemption Status';
export const PLAN_DATE = 'Plan Date';
export const PRIMARY_AGREEMENT_HOLDER = 'Primary Agreement Holder';
export const OTHER_AGREEMENT_HOLDER = 'Agreement Holder (Other)';

// Range use plan pastures
export const ALLOWABLE_AUMS = 'Allowable AUMs';
export const PRIVATE_LAND_DEDUCTION = 'Private Land Deduction (%)';
export const GRACE_DAYS = 'Grace Days';
export const PASTURE_NOTES = 'Pasture Notes';

// Range use plan schedules
export const PASTURE = 'Pasture';
export const LIVESTOCK_TYPE = 'Livestock Type';
export const NUM_OF_ANIMALS = '# of Animals';
export const DATE_IN = 'Date in';
export const DATE_OUT = 'Date out';
export const DAYS = 'Days';
export const PLD = 'PLD';
export const CROWN_AUMS = 'Crown AUMs';

export const CONTACT_NO_EXIST = 'Contact doesn\'t exist';

// messages
export const UNEXPECTED_ERROR = 'An unexpected error occurred.';
export const STATUS404 = 'The request is currently not available, please try later.';
export const STATUS500 = 'Internal server error occurred, please contact the administrator.';
export const UPDATE_PLAN_STATUS_SUCCESS = 'You have successfully updated the status of the range use plan.';
export const UPDATE_AGREEMENT_ZONE_SUCCESS = 'You have successfully updated the zone of the range use plan.';
export const ASSIGN_STAFF_TO_ZONE_SUCCESS = 'You have successfully assigned the staff to the zone.';
export const SAVE_PLAN_AS_DRAFT_SUCCESS = 'You have successfully saved the range use plan as a draft.';
export const SUBMIT_PLAN_SUCCESS = 'You have successfully submitted the range use plan to the range staff.';
export const EMPTY_GRAZING_SCHEDULE_ENTRIES = 'Schedule must have at least 1 entry.';
export const INVALID_GRAZING_SCHEDULE_ENTRY = 'Schedule has one or more invalid entries.';
export const TOTAL_AUMS_EXCEEDS = 'Total AUMs exceeds authorized AUMs.';
export const USER_NOT_ACTIVE = 'This account is not active yet, please contact the administrator.';
export const USER_NO_ROLE = 'This account doesn\'t have a role, please contact the administrator.';
export const USER_NOT_REGISTERED = 'This account has not been registered.';
export const LOADING_USER = 'Loading User Information';
export const LINK_CLIENT_SUCCESS = 'You have successfully linked the client.';
export const NO_INTERNET = 'There is no Internet connection.';
export const TYPE_CLIENT_NAME = 'Type name of the client';
export const NO_RESULTS_FOUND = 'No result founds.';
export const ERROR_OCCUR = 'Error Occured!';

// modals
export const COMPLETED_CONFIRMATION_HEADER = 'Update Status: Completed';
export const COMPLETED_CONFIRMATION_CONTENT = 'COMPLETE indicates that a RUP has either been APPROVED or DISCARDED. If you change status to COMPLETE you will no longer be able to make edits to this RUP. Would you like to switch this RUP to complete?';
export const PENDING_CONFIRMATION_HEADER = 'Update Status: Pending';
export const PENDING_CONFIRMATION_CONTENT = 'PENDING indicates that a RUP is in edit mode. It is used during initial creation if the decision maker has requested edits before approving. Do not switch the status to PENDING unless the decision maker has requested specific edits. Would you like to switch this RUP to Pending?';
export const UPDATE_CONTACT_CONFIRMATION_HEADER = 'Confirmation: Update Contact';
export const UPDATE_CONTACT_CONFIRMATION_CONTENT = 'Are you sure you want to update the contact?';
export const DELETE_SCHEDULE_FOR_AH_HEADER = 'Confirmation: Deleting Grazing Schedule';
export const DELETE_SCHEDULE_FOR_AH_CONTENT = 'This schedule will be permanently deleted. Are you sure you want to delete this schedule?';
export const DELETE_SCHEDULE_ENTRY_FOR_AH_HEADER = 'Confirmation: Deleting Grazing Schedule Entry';
export const DELETE_SCHEDULE_ENTRY_FOR_AH_CONTENT = 'This schedule entry will be permanently deleted. Are you sure you want to delete this schedule entry?';
export const UPDATE_CLIENT_ID_FOR_AH_HEADER = 'Confirmation: Link Client';
export const UPDATE_CLIENT_ID_FOR_AH_CONTENT = 'Are you sure you want to link this user to the client?';
export const SUBMIT_RUP_CHANGE_FOR_AH_HEADER = 'Ready to Submit?';
export const SUBMIT_RUP_CHANGE_FOR_AH_CONTENT = 'Once submitted you can no longer make edits to the Range Use Plan. Do not submit until you are satisfied with all content.';

// banners
export const DETAIL_RUP_BANNER_CONTENT = 'View the full PDF file or update the status of the range use plan.';
export const DETAIL_RUP_EDIT_BANNER_CONTENT = 'Please finalize your changes and submit for final approval';
export const SELECT_RUP_BANNER_HEADER = 'Select Range Use Plan';
export const SELECT_RUP_BANNER_CONTENT = 'View details of each range use plan. Enter RAN number, agreement holder\'s name, or staff contact in the search box to search for a specific range use plan.';
export const MANAGE_ZONE_BANNER_HEADER = 'Manage Zone';
export const MANAGE_ZONE_BANNER_CONTENT = 'Follow steps to assign a zone from the current staff to other staff.';
export const RUP_CREATED_FOR_AH_CONTENT = 'Please confirm your range use plan.';
export const RUP_IN_DRAFT_FOR_AH_CONTENT = 'Please finalize your changes and submit for Range staff review.';
export const RUP_PENDING_FOR_AH_CONTENT = 'Your range use plan is currently being reviewed by range staff.';
export const RUP_CHANGE_REQUESTED_FOR_AH_CONTENT = 'Your range use plan was reviewed by Range staff and requires revisions. Please make changes and resubmit.';
export const RUP_COMPLETE_FOR_AH_CONTENT = 'Your range use plan is approved by Range staff.';
export const MANAGE_CLIENT_BANNER_HEADER = 'Manage Client';
export const MANAGE_CLIENT_BANNER_CONTENT = 'Follow the steps below to link the user to their client number.';

// statuses of range use plan in user perspective
export const INPUT_REQUIRED = 'Input Required';
export const REVIEW_REQUIRED = 'Review Required';
export const IN_REVIEW = 'In Review';
export const REVISIONS_REQUESTED = 'Revisions Requested';
export const SENT_FOR_INPUT = 'Sent for Input';
export const IN_PROGRESS = 'In Progress';
