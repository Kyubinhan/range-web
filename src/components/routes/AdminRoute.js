import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { LOGIN } from '../../constants/routes';
import LandingPage from '../LandingPage';
import User from '../../models/User';

const propTypes = {
  component: PropTypes.func.isRequired,
  path: PropTypes.string.isRequired,
  user: PropTypes.shape({}),
};

const defaultProps = {
  user: null,
};

const AdminRoute = ({ component: Component, user: u, ...rest }) => (
  <Route
    {...rest}
    render={(props) => {
        const user = new User(u);
        if (user.isAdmin) {
          return <LandingPage {...props} component={Component} />;
        }
        return <Redirect to={LOGIN} />;
      }
    }
  />
);

AdminRoute.propTypes = propTypes;
AdminRoute.defaultProps = defaultProps;
export default AdminRoute;
