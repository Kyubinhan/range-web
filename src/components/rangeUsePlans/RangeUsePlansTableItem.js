import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { RANGE_USE_PLAN } from '../../constants/routes';
import { Table } from 'semantic-ui-react';
import { withRouter } from 'react-router-dom';
import { Status } from '../common';

const propTypes = {
  rangeUsePlan: PropTypes.object.isRequired,
};

const defaultProps = {
  rangeUsePlan: {},
};

export class RangeUsePlansTableItem extends Component {
  onRowClicked = () => {
    const { rangeUsePlan, history } = this.props;
    history.push(`${RANGE_USE_PLAN}/${rangeUsePlan.id}`)
  }

  render() {
    const { rangeUsePlan } = this.props;

    return (
      <Table.Row
        className="range-use-plans-table-item"
        onClick={this.onRowClicked}
      >
        <Table.Cell>{`RAN07123${rangeUsePlan.number}`}</Table.Cell>
        <Table.Cell>{rangeUsePlan.region}</Table.Cell>
        <Table.Cell>{rangeUsePlan.tenureHolder.name}</Table.Cell>
        <Table.Cell>{rangeUsePlan.rangeOfficer.name}</Table.Cell>
        <Table.Cell><Status status={rangeUsePlan.status}/></Table.Cell>
      </Table.Row>
    );
  }
}

RangeUsePlansTableItem.propTypes = propTypes;
RangeUsePlansTableItem.defaultProps = defaultProps;
export default withRouter(RangeUsePlansTableItem);