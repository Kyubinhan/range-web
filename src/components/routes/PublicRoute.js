import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { RANGE_USE_PLANS } from '../../constants/routes';

const propTypes = {
  component: PropTypes.func.isRequired,
  path: PropTypes.string.isRequired,
  user: PropTypes.shape({}),
};

const defaultProps = {
  user: null,
};

/**
 * If we have a logged-in user, redirect to the home page. Otherwise, display the component.
 */
const PublicRoute = ({ component: Component, user, ...rest }) => (
  <Route
    {...rest}
    render={(props) => {
      if (user) {
        return <Redirect to={RANGE_USE_PLANS} />;
      }
      return <Component {...props} />;
    }}
  />
);

PublicRoute.propTypes = propTypes;
PublicRoute.defaultProps = defaultProps;
export default PublicRoute;
