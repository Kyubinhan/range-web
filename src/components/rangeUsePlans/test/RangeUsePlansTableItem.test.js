import React from 'react';
import { shallow } from 'enzyme';
import { RangeUsePlansTableItem } from '../RangeUsePlansTableItem';
import { getMockRangeUsePlan } from './mockValues';
import { RANGE_USE_PLAN } from '../../../constants/routes';

const props = {};
const setupProps = () => {
  props.rangeUsePlan = getMockRangeUsePlan(1);
  props.history = {
    push: jest.fn(),
  }
};

beforeEach(() => {
  setupProps();
});

describe('RangeUsePlansTableItem', () => {
  xit('renders correctly', () => {
    const wrapper = shallow(<RangeUsePlansTableItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('push to the new route when clicking on Table.row', () => {
    const wrapper = shallow(<RangeUsePlansTableItem {...props} />);
    wrapper.simulate('click', {});

    expect(props.history.push).toHaveBeenCalledTimes(1);
    expect(props.history.push).toHaveBeenCalledWith(`${RANGE_USE_PLAN}/${props.rangeUsePlan.id}`);
  });

  describe('Event handlers', () => {
    it('onRowClicked is called to push to the new route', () => {
      const wrapper = shallow(<RangeUsePlansTableItem {...props} />);
      wrapper.instance().onRowClicked();

      expect(props.history.push).toHaveBeenCalledTimes(1);
      expect(props.history.push).toHaveBeenCalledWith(`${RANGE_USE_PLAN}/${props.rangeUsePlan.id}`);
    });
  });
});