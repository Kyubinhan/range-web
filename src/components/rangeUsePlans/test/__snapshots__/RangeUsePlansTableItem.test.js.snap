// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeUsePlansTableItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RangeUsePlansTableItem
    history={
        Object {
            "push": [Function],
          }
    }
    rangeUsePlan={
        Object {
            "id": "RAN071231",
            "number": 1,
            "rangeOfficer": Object {
              "name": "Obi-Wan Kenobi",
            },
            "region": "Victoria Ranch 1",
            "status": "Completed",
            "tenureHolder": Object {
              "name": "Luke Skywalker",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "tr",
      "cellAs": "td",
      "children": Array [
        <TableCell
          as="td"
>
          RAN071231
</TableCell>,
        <TableCell
          as="td"
>
          Victoria Ranch 1
</TableCell>,
        <TableCell
          as="td"
>
          Luke Skywalker
</TableCell>,
        <TableCell
          as="td"
>
          Obi-Wan Kenobi
</TableCell>,
        <TableCell
          as="td"
>
          <Unknown
                    status="Completed"
          />
</TableCell>,
      ],
      "className": "range-use-plans-table-item",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "td",
          "children": "RAN071231",
        },
        "ref": null,
        "rendered": "RAN071231",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "td",
          "children": "Victoria Ranch 1",
        },
        "ref": null,
        "rendered": "Victoria Ranch 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "td",
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "td",
          "children": "Obi-Wan Kenobi",
        },
        "ref": null,
        "rendered": "Obi-Wan Kenobi",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "td",
          "children": <Unknown
            status="Completed"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "status": "Completed",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "tr",
        "cellAs": "td",
        "children": Array [
          <TableCell
            as="td"
>
            RAN071231
</TableCell>,
          <TableCell
            as="td"
>
            Victoria Ranch 1
</TableCell>,
          <TableCell
            as="td"
>
            Luke Skywalker
</TableCell>,
          <TableCell
            as="td"
>
            Obi-Wan Kenobi
</TableCell>,
          <TableCell
            as="td"
>
            <Unknown
                        status="Completed"
            />
</TableCell>,
        ],
        "className": "range-use-plans-table-item",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "td",
            "children": "RAN071231",
          },
          "ref": null,
          "rendered": "RAN071231",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "td",
            "children": "Victoria Ranch 1",
          },
          "ref": null,
          "rendered": "Victoria Ranch 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "td",
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "td",
            "children": "Obi-Wan Kenobi",
          },
          "ref": null,
          "rendered": "Obi-Wan Kenobi",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "td",
            "children": <Unknown
              status="Completed"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "status": "Completed",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
